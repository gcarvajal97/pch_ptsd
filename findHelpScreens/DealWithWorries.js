import React, { Component } from "react";
import { View, StyleSheet, Text, ScrollView } from "react-native";
import { Video } from "expo-av";
import translate, { getVideo } from "../components/translateService";
import colors from '../colors';

const videos = {
    en: require("../assets/videos/vidDealWithWorries.mp4"),
    es: require("../assets/videos/vidDealWithWorriesEs.mp4"),
};

class DealWithWorries extends Component {
    // Gives the NavBar a title for this specific screen
    static navigationOptions = {
        headerTitle: translate("dealWithWorries.header")
    };

    render() {
        return (
            <View style={styles.container}>
                <Text
                    style={{
                        backgroundColor: colors.accent,
                        color: colors.primary.white,
                        fontFamily: "avenir-heavy",
                        fontSize: 22,
                        textAlign: "center",
                        alignSelf: "stretch"
                    }}
                >
                    {translate("dealWithWorries.video_title")}
                </Text>
                <Video
                    source={getVideo(videos)}
                    rate={1.0}
                    volume={1.0}
                    resizeMode={Video.RESIZE_MODE_COVER}
                    shouldPlay
                    useNativeControls
                    style={{
                        height: 204,
                        width: "100%",
                        maxHeight: 254,
                        borderColor: colors.accent,
                        borderWidth: 14,
                        borderTopWidth: 0,
                        borderBottomWidth: 0,
                        marginBottom: -1
                    }}
                    accessibilityLabel={translate(
                        "dealWithWorries.accessibilityLabel"
                    )}
                    accessibilityHint={translate(
                        "dealWithWorries.accessibilityHint"
                    )}
                />
                <Text
                    style={{
                        backgroundColor: colors.accent,
                        color: colors.primary.white,
                        alignSelf: "stretch",
                        paddingLeft: 15,
                        fontSize: 15,
                        borderBottomWidth: 2,
                        fontFamily: "avenir-medium",
                        borderBottomColor: colors.accent
                    }}
                >
                    <Text style={{ fontFamily: "avenir-heavy" }}>
                        {translate("dealWithWorries.video_subtitle_1")}
                    </Text>
                    {translate("dealWithWorries.video_subtitle_2")}
                </Text>
                <ScrollView>
                    <Text style={styles.headline}>
                        {translate("dealWithWorries.q1")}
                    </Text>
                    <Text style={styles.paragraph}>
                        {translate("dealWithWorries.q1_p1")}
                    </Text>
                    <Text style={styles.paragraph}>
                        {translate("dealWithWorries.q1_p2")}
                    </Text>

                    <Text style={styles.headline}>
                        {translate("dealWithWorries.q2")}
                    </Text>
                    <Text style={styles.paragraph}>
                        {translate("dealWithWorries.q2_p1")}
                    </Text>
                    <Text style={styles.paragraph}>
                        {translate("dealWithWorries.q2_p2")}
                    </Text>
                    <Text style={styles.paragraph}>
                        {translate("dealWithWorries.q2_p3")}
                    </Text>

                    <Text style={styles.headline}>
                        {translate("dealWithWorries.q3")}
                    </Text>
                    <Text style={styles.paragraph}>
                        {translate("dealWithWorries.q3_p1")}
                    </Text>
                    <Text style={styles.paragraph}>
                        {translate("dealWithWorries.q3_p2")}
                    </Text>

                    <Text style={styles.headline}>
                        {translate("dealWithWorries.q4")}
                    </Text>
                    <Text style={styles.paragraphBold}>
                        {translate("dealWithWorries.q4_tip1")}
                    </Text>
                    <Text style={styles.bulletLight}>
                        {translate("dealWithWorries.q4_tip1_bullet")}
                    </Text>
                    <Text style={styles.paragraphBold}>
                        {translate("dealWithWorries.q4_tip2")}
                    </Text>
                    <Text style={styles.bulletLight}>
                        {translate("dealWithWorries.q4_tip2_bullet")}
                    </Text>
                    <Text style={styles.paragraphBold}>
                        {translate("dealWithWorries.q4_tip3")}
                    </Text>
                    <Text style={styles.bulletLight}>
                        {translate("dealWithWorries.q4_tip3_bullet")}
                    </Text>
                    <Text style={styles.paragraphBold}>
                        {translate("dealWithWorries.q4_tip4")}
                    </Text>
                    <Text style={styles.bulletLight}>
                        {translate("dealWithWorries.q4_tip4_bullet")}
                    </Text>
                    <Text style={styles.paragraphBold}>
                        {translate("dealWithWorries.q4_tip5")}
                    </Text>
                    <Text style={styles.bulletLight}>
                        {translate("dealWithWorries.q4_tip5_bullet1")}
                    </Text>
                    <Text style={styles.bulletLight}>
                        {translate("dealWithWorries.q4_tip5_bullet2")}
                    </Text>

                    <Text style={styles.headline}>
                        {translate("dealWithWorries.q5")}
                    </Text>
                    <Text style={styles.paragraph}>
                        {translate("dealWithWorries.q5_p1")}
                    </Text>
                    <Text style={styles.paragraph}>
                        {translate("dealWithWorries.q5_p2")}
                    </Text>
                </ScrollView>
            </View>
        );
    }
}
export default DealWithWorries;

const styles = StyleSheet.create({
    container: {
        flex: 1,
        alignItems: "center",
        margin: 10,
        backgroundColor: colors.primary.white,
        shadowColor: colors.primary.black,
        shadowOffset: {
            width: 0,
            height: 2
        },
        shadowOpacity: 0.23,
        shadowRadius: 2.62,

        elevation: 4
    },
    bullet: {
        marginVertical: 1,
        marginHorizontal: 18,
        textAlign: "left",
        alignSelf: "stretch",
        fontFamily: "avenir-heavy",
        fontSize: 14,
        paddingTop: 4
    },
    bulletLight: {
        marginVertical: 1,
        marginHorizontal: 18,
        textAlign: "left",
        alignSelf: "stretch",
        fontFamily: "avenir-medium",
        fontSize: 14,
        paddingTop: 4
    },
    paragraph: {
        alignSelf: "center",
        paddingVertical: 1,
        paddingHorizontal: 5,
        fontSize: 14,
        fontFamily: "avenir-medium",
        marginTop: 5
    },
    paragraphBold: {
        textAlign: "left",
        paddingVertical: 1,
        paddingHorizontal: 5,
        fontSize: 14,
        marginTop: 5,
        fontFamily: "avenir-heavy"
    },
    headline: {
        fontFamily: "avenir-heavy",
        fontSize: 20,
        alignSelf: "stretch",
        textAlign: "left",
        marginLeft: 5,
        marginTop: 4,
        marginBottom: 3,
        color: colors.accent
    }
});
