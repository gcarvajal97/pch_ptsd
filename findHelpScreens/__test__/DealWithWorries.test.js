import React from 'react';
import DealWithWorries from '../DealWithWorries';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
    const tree = renderer.create(<DealWithWorries/>).toJSON();
    expect(tree).toMatchSnapshot();
}) 