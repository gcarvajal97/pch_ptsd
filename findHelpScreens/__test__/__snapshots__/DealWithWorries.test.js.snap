// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DealWithWorries renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "elevation": 4,
      "flex": 1,
      "margin": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.23,
      "shadowRadius": 2.62,
    }
  }
>
  <Text
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#2089DC",
        "color": "white",
        "fontSize": 22,
        "fontWeight": "bold",
        "textAlign": "center",
      }
    }
  >
    Dealing With Fears and Worries
  </Text>
  <View
    pointerEvents="box-none"
    style={
      Object {
        "borderBottomWidth": 0,
        "borderColor": "#2089DC",
        "borderTopWidth": 0,
        "borderWidth": 14,
        "height": 204,
        "marginBottom": -1,
        "maxHeight": 254,
        "overflow": "hidden",
        "width": "100%",
      }
    }
  >
    <ViewManagerAdapter_ExpoVideoView
      accessibilityHint="Video on dealing with new fears and worries"
      accessibilityLabel="Video on dealing with new fears and worries"
      proxiedProperties={
        Object {
          "onError": [Function],
          "onFullscreenUpdate": [Function],
          "onLoad": [Function],
          "onLoadStart": [Function],
          "onReadyForDisplay": [Function],
          "onStatusUpdate": [Function],
          "resizeMode": "mock",
          "source": Object {
            "headers": undefined,
            "overridingExtension": null,
            "uri": "uri",
          },
          "status": Object {
            "rate": 1,
            "shouldPlay": true,
            "volume": 1,
          },
          "useNativeControls": true,
        }
      }
      style={
        Object {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
  </View>
  <Text
    style={
      Object {
        "alignSelf": "stretch",
        "backgroundColor": "#2089DC",
        "borderBottomColor": "#2089DC",
        "borderBottomWidth": 2,
        "color": "white",
        "fontSize": 15,
        "paddingLeft": 15,
      }
    }
  >
    <Text
      style={
        Object {
          "fontWeight": "bold",
        }
      }
    >
      New fears or worries? 
    </Text>
    Watch the video and read below to learn more about these symptoms.
  </Text>
  <RCTScrollView>
    <View>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#2089DC",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 3,
            "marginLeft": 5,
            "marginTop": 4,
            "textAlign": "left",
          }
        }
      >
        How do new fears develop?
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        When something scary happens, we often learn to treat things connected to it as if they were danger signals. The next time something reminds us of what happened, our body reacts as if we were in danger, even if we are safe.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        A trauma reminder can be anything – a place or a person, a sight or a sound, or even a smell. It might be something you hardly notice, till the scared feelings come flooding back.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#2089DC",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 3,
            "marginLeft": 5,
            "marginTop": 4,
            "textAlign": "left",
          }
        }
      >
        How do we get over new fears - dealing with emotional triggers?
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        In managing emotional triggers, it can really help to face reminders over and over, in the course of normal life. We can feel very anxious at first, but each time we experience a trauma reminder and hang in there long enough to know that it does not lead to danger, the scared feelings tend to get weaker.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        It is natural for your child to want to stay away from things that remind and upset him. But too much avoidance can get in the way of the natural recovery process, and make it harder to get over feeling scared.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        As a parent, you play an important role in helping your child face new fears without getting stuck in avoidance.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#2089DC",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 3,
            "marginLeft": 5,
            "marginTop": 4,
            "textAlign": "left",
          }
        }
      >
        How can parents tell if fears or worries are really a problem?
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        Ask yourself: Does feeling afraid or nervous stop your child from getting back to normal safe activities? Or stop him from enjoying things even if he manages to do them?
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        If so, you can help your child deal with new fears and worries before they become a bigger problem.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#2089DC",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 3,
            "marginLeft": 5,
            "marginTop": 4,
            "textAlign": "left",
          }
        }
      >
        What can parents do?
      </Text>
      <Text
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
            "textAlign": "left",
          }
        }
      >
        1. Help your child make changes that really do help keep him safe.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "paddingTop": 4,
            "textAlign": "left",
          }
        }
      >
         - Like wearing a helmet, crossing the street carefully, or telling you where he is going.
      </Text>
      <Text
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
            "textAlign": "left",
          }
        }
      >
        2. Be a 'reality check' for your child.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "paddingTop": 4,
            "textAlign": "left",
          }
        }
      >
         - If your child is nervous about doing something, talk together and help her figure out: 'Is this really helping me stay safe?' OR 'Am I avoiding it just so I will not feel nervous?'
      </Text>
      <Text
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
            "textAlign": "left",
          }
        }
      >
        3. Help your child understand 'trauma reminders' emotional triggers and avoidance.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "paddingTop": 4,
            "textAlign": "left",
          }
        }
      >
         - Sometimes kids need help to figure out what makes them feel scared or nervous.
      </Text>
      <Text
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
            "textAlign": "left",
          }
        }
      >
        4. Encourage your child.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "paddingTop": 4,
            "textAlign": "left",
          }
        }
      >
         - Help him keep doing those safe things that make him nervous -- and praise his courage when he does!
      </Text>
      <Text
        style={
          Object {
            "fontSize": 14,
            "fontWeight": "bold",
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
            "textAlign": "left",
          }
        }
      >
        5. Notice your own worries.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "paddingTop": 4,
            "textAlign": "left",
          }
        }
      >
         - Ask yourself: Do my worries stop me from letting my child do things he really should be allowed to do?
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "fontSize": 14,
            "marginHorizontal": 18,
            "marginVertical": 1,
            "paddingTop": 4,
            "textAlign": "left",
          }
        }
      >
         - Use a trusted friend or family member as a “reality check” for yourself – to help you sort through your worries about your child’s safety.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "stretch",
            "color": "#2089DC",
            "fontSize": 20,
            "fontWeight": "bold",
            "marginBottom": 3,
            "marginLeft": 5,
            "marginTop": 4,
            "textAlign": "left",
          }
        }
      >
        When to get more help
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        If you are not sure whether your child's worries or fears are realistic, or how to encourage your child to face things that he is avoiding, get some help.
      </Text>
      <Text
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 14,
            "marginTop": 5,
            "paddingHorizontal": 5,
            "paddingVertical": 1,
          }
        }
      >
        Talk with your family doctor, school counselor, or someone else you trust, or seek a counselor who can help you and your child in managing emotional triggers and with traumatic stress reactions.
      </Text>
    </View>
  </RCTScrollView>
</View>
`;
